### PR TITLE
specifications/issues/597 for persistence/issues/391 TCK Challenge for Wrong typed comparison expressions in TCK

### PR DESCRIPTION
**Pull request by scottmarlow**
https://github.com/jakartaee/specifications/issues/597 is to address the https://github.com/jakartaee/persistence/issues/391 Persistence 3.1 TCK challenge, please promote staged TCKs:
- http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-persistence-tck-3.1.2.zip
- https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-jakartaeetck-10.0.2.zip

I verified that the staged TCKs (http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-persistence-tck-3.1.2.zip + https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-jakartaeetck-10.0.2.zip) are updated to exclude the challenged tests.

CI results are in https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-standalonetck-build-run-100/116 for http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl//jakarta-persistence-tck-3.1.2.zip and https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/10.0.x/48/testReport/com.sun.ts.tests.jpa.core.query.apitests/Client for https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-jakartaeetck-10.0.2.zip

### CODE REVIEW

**Comment by scottmarlow** [comment]
The preview page https://deploy-preview-598--jakartaee-specifications.netlify.app/ looks right to me.  Someone needs to run the script to promote the two mentioned staged TCKs and then this should be ready to merge (pending review from one other).

**Review by jeanouii — APPROVED** [review]
The diff looks good and I clicked the link as well.
I think we are good to merge

**Pull request merged**
